polar-core/src/sources.rs

```diff
@@ -0,0 +1,66 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Hash)]
+pub enum SourceInfo {
+    // From the parser
+    Parser {
+        /// Index into the source map stored in the knowledge base
+        src_id: u64,
+
+        /// Location of the term within the source map
+        left: usize,
+        right: usize,
+    },
+
+    /// Created as a temporary variable
+    TemporaryVariable,
+
+    /// From an FFI call
+    Ffi,
+
+    /// Created for a test
+    Test,
```

Comment:
should we `#[cfg(test)]` this variant?

Reply:
That's an interesting question. The layout of the `enum` would then be different in test & prod, which I suppose shouldn't worry me but does. Does anyone else have opinions?

Reply:
#TrustInRustC

More seriously, I think Rust intentionally makes no guarantees about [layout for enums using the default representation](https://doc.rust-lang.org/reference/type-layout.html#representations), so I'd be shocked if different layouts led to different runtime characteristics.